lib/tasks/cleanup.rake

```diff
@@ -57,6 +57,13 @@ namespace :cleanup do
     end
   end
 
+  desc 'Reindex banned users'
+  task :reindex_banned_users => :environment do
+    User.where("ban_text <> ''").find_each do |user|
+      user.xapian_mark_needs_index
+    end
+  end
+
```

Comment:
That's what I've used.

The problem is that this just creates records in the `acts_as_xapian_jobs` table. `ActsAsXapian.update_index` is called every 5 mins through a cron job (via a script and rake task), which then tries to process _all_ records in the `acts_as_xapian_jobs` table. I don't know what happens if a job hasn't completed by the time the next one starts.

Maybe the lock file just prevents any new indexes happening until the "big" job has finished? What happens if new records are added to `acts_as_xapian_jobs` while the big job is running  it might never exit?

IDK – thinking out loud a bit here :)